protos/shared/shared.proto

```diff
@@ -7,6 +7,10 @@ option go_package = "shared";
 import "github.com/gogo/protobuf/gogoproto/gogo.proto";
 import "google/protobuf/timestamp.proto";
 
+service RuntimeService {
+  rpc CheckRuntime(RuntimeInfoRequest) returns (RuntimeInfoReport);
+}
+
 message RuntimeInfoRequest {
   bytes request_address = 1;
```

Comment:
This request address would be the public key, otherwise you need to add public key for the signature.

Reply:
ok.